### ISSUE TITLE
HTTP: configuration option for a Custom CA

### ISSUE THREAD

**Issue by pvogler** [kuskoman/logstash-exporter#359]
### Feature Description

With issue report #337 HTTP tls verification for v1 was implemented.
In  fact this feature is missing in version 2 and the master branch.

Easy way would be to upstream the verification skipping, but I would prefer to allow an option to configure a custom ca or disable verification manually.

### Version of logstash-exporter this feature request applies to

v2

### Motivation and Goals

Motivation is, to get the exporter working out of the box also with the ECK operator deployment.

### Alternatives Considered

Disabling TLS on Logstash side would work as well, but from security perspective this is not a good way.

**Comment by satk0** [comment]
Good catch, I could rewrite it for v2

**Comment by pvogler** [comment]
If you provide a nightly image, I would be happy to test it for you.

**Comment by satk0** [comment]
Thanks in advance, it would sure be a huge help! Right now, I am still waiting for @kuskoman to finally merge my `hot-reload` PR :face_exhaling:  

I honestly can't wait to start working on it, I have the right mood and motivation to do it :smile:

**Comment by satk0** [comment]
Done #382

**Comment by pvogler** [comment]
I will provide feedback upcoming week.After I got back from vacation.On 30. Oct 2024, at 23:08, Adam Satko ***@***.***> wrote:﻿
Done #382

**Comment by satk0** [comment]
Chill, relax on your holidays ;) 

We first have to provide a nightly image for that and merge the PR. 
@kuskoman do you have time for that?

**Comment by satk0** [comment]
@pvogler Did you managed to check it? If it works, I could just close this issue heh :sweat_smile:

**Comment by pvogler** [comment]
@satk0 Looks like the direct port from version. As a workaround the possibility to disable the TLS verification is a good starting point and does work as expected. But to transform this exporter in a really secure one, it should be possible to specify a custom ca file provide to the exporter.
e.g. as config parameter
Logstash:
  customer: "path to custom ca file"
  
  If you think this would be a good option as well, we could also close this issue and open a new one, concentrating on the custom ca.
  
  And sorry for the late reply. Got some different business to finish first.

**Comment by satk0** [comment]
Yeah, it is indeed a workaround. I see what you mean, for me it looks like a proper improvement. @kuskoman wdyt?

**Comment by pvogler** [comment]
As I said. For me it is also valid to proceed like this, as a first step. Thanks for your effort.On 29. Nov 2024, at 18:56, Adam Satko ***@***.***> wrote:﻿
Yeah, it is indeed a workaround. I see what you mean, for me it looks like a proper improvement. @kuskoman wdyt?

**Comment by satk0** [comment]
@pvogler me and kuskoman agreed on this idea, and I'd be working on it. Could you just let me know, is it about setting specifically root certificate authority (RootCA)?
I am not an expert in this field and your response would be really helpful, idk where to start with it hah

**Comment by kuskoman** [comment]
i feel like this got duplicated by #393
@satk0 please take a look on the other issue, some user already implemented it, but did not create a PR

**Comment by satk0** [comment]
@kuskoman You mean this? https://github.com/chanyongkit/logstash-exporter/commit/3f7679adef164d1a677503f1c226ed443658a105

This implements HTTP TLS but I think there is the need to specify custom root ca to tls.config for each logstash instance instead of merely using HttpInsecure var, wdyt?

**Comment by kuskoman** [comment]
yep, butni believe this needs some testing, you can copy the branch to your repo, so that the commit will have its original author

**Comment by satk0** [comment]
I will just rewrite it, I am not that git-savy :stuck_out_tongue:

**Comment by satk0** [comment]
Do we need this commit for v1, or should I implement it only for v2?

**Comment by pvogler** [comment]
@satk0 @kuskoman
Just for clarification: #393 adds support for encrypting the exporter, so that Prometheus talks https with this exporter.

I want to talk TLS with the Logstash API. So 393 is not a duplicate
And for these calls I want to specify a RootCA-file. One file should be enough. Either all CAs can be put inside this file like the ca-bundle.crt e.g. on Redhat, or we simply start multiple exporters if different CAs are required.

And for me v2 would be sufficient.

Does this help in clarification?

**Comment by satk0** [comment]
Yeah, sure it helps, thanks for being so active ;) 

I've already started working here #401 on encrypting the exporter, when I finish writing tests for it, I will start resolving this one so stay tuned :smile_cat:

**Comment by kuskoman** [comment]
Check the attached PR, i made some breaking changes for v2, but it should be more configurable now (#428)
